### ISSUE TITLE
Reproduction Issues

### ISSUE THREAD

**Issue by JianhuiWei7** [NVIDIA-NeMo/ProRL-Agent-Server#40]
Hi, I am recently reproducing the training example of `examples/swegym_slime_grpo`

I have encountered a few problems, and I wish to discuss with you.

1. The training dynamic, grad_norm is very large. Here I paste the log of step 1 and 28
```
step 1: {'train/loss': 0.10215942561626434, 'train/pg_loss': 0.10215942561626434, 'train/entropy_loss': 0.6176612973213196, 'train/pg_clipfrac': 0.0, 'train/ppo_kl': 0.0, 'train/train_rollout_logprob_abs_diff': 0.030350876972079277, 'train/kl_loss': -7.420481763775744e-12, 'train/ois': 0.875, 'train/tis': 0.8740883469581604, 'train/tis_clipfrac': 0.0006311089382506907, 'train/tis_abs': 0.02566334791481495, 'train/grad_norm': 2955012562.8240876, 'train/lr-pg_0': 1e-07, 'train/lr-pg_1': 1e-07, 'train/step': 1}

step 28: {'train/loss': 0.1451725959777832, 'train/pg_loss': 0.14517192542552948, 'train/entropy_loss': 0.3434755206108093, 'train/pg_clipfrac': 0.0, 'train/ppo_kl': 0.0, 'train/train_rollout_logprob_abs_diff': 0.04285452514886856, 'train/kl_loss': 0.0007035313756205142, 'train/ois': 0.75, 'train/tis': 0.7496934533119202, 'train/tis_clipfrac': 0.0009880214929580688, 'train/tis_abs': 0.029075611382722855, 'train/grad_norm': 2217390918.482301, 'train/lr-pg_0': 1e-07, 'train/lr-pg_1': 1e-07, 'train/step': 28}
```
loss seems to be normal, as well as the other values, but the grad_norm is unexpectedly large. And I still didn;t figure out why this may occur.

2. The training sometimes may occur NaN grad in the backpropagation， Is this something to do with the large grad_norm

3. If luckily, The whole 74 training steps finished without reporting NaN, the `polar/reward_mean` in the launch_e2e-\<time_stamp\>.log is not increasing as shown in the README.md figure. I run with the codex harness. The metric shows frequent jumps. I was thinking maybe. this is the training phase, each step, the model sees a different questions, so the metrics may be jumping. And the code didn't evaluate on the test set on a fixed steps of optimization. So we can not tell the overall accuracy gain on the test sets. If this is the case, how can i add a test set after like 5 steps of optimization?

Finally, thank you for the framework.

**Comment by JianhuiWei7** [comment]
I figure it out that it is attributed to the training backend used. Since the launch_e2e.sh skips flash-attn install for machines that are not SM100，without this backend, it will turn to use TE's default engine, which gives poor support. And my machine is SM90, and I build flash-attn, and use it, the grad_norm becomes normal value (within 10)
